src/libsync/theme.cpp

```diff
@@ -129,39 +158,51 @@ QString Theme::configFileName() const
 
 QIcon Theme::applicationIcon() const
 {
-    return themeIcon(QStringLiteral(APPLICATION_ICON_NAME "-icon"));
+    return themeUniversalIcon(QStringLiteral(APPLICATION_ICON_NAME "-icon"));
 }
 
 QIcon Theme::aboutIcon() const
 {
     return applicationIcon();
 }
 
-bool Theme::isUsingDarkTheme(IconType type) const
+bool Theme::isUsingDarkTheme() const
 {
-    if (!_hasDarkColoredTheme && type != IconType::VanillaIcon) {
-        return false;
-    }
-    return QPalette().base().color().lightnessF() <= 0.5;
+    //TODO: replace by a command line switch
+    static bool forceDark = qEnvironmentVariableIntValue("OWNCLOUD_FORCE_DARK_MODE") != 0;
+    return forceDark || QPalette().base().color().lightnessF() <= 0.5;
+}
+
+
+QIcon Theme::themeUniversalIcon(const QString &name, Theme::IconType iconType) const
+{
+    return loadIcon(QStringLiteral("universal"), name, iconType);
+}
+
+QIcon Theme::themeTrayIcon(const QString &name, bool sysTrayMenuVisible, IconType iconType) const
+{
+    auto &icon = loadIcon(systrayIconFlavor(_mono, sysTrayMenuVisible), name, iconType);
+#ifdef Q_OS_MAC
+    // This defines the icon as a template and enables automatic macOS color handling
+    // See https://bugreports.qt.io/browse/QTBUG-42109
+    icon.setIsMask(_mono && !sysTrayMenuVisible);
```

Comment:
Maybe worth checking whether it's correct to use `setIsMask` on the reference or if it is better to use it on the copy that is returned later on.